Angular Client fails to compile With more than one entity

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

Generating an Angular client on 6.9.0 with entities results in:
```
ERROR in src/main/webapp/app/entities/entity.module.ts:6:27 - error TS2345: Argument of type '({ path: string; loadChildren: () => Promise<any>; } | undefined)[]' is not assignable to parameter of type 'Routes'.
  Type '{ path: string; loadChildren: () => Promise<any>; } | undefined' is not assignable to type 'Route'.
    Type 'undefined' is not assignable to type 'Route'.

  6     RouterModule.forChild([
                              ~
  7       {
    ~~~~~~~
... 
 16       /* jhipster-needle-add-entity-route - JHipster will add entity modules routes here */
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 17     ]),
    ~~~~~
```

##### **Reproduce the error**

1. Generate a new Monolith app
2. Add an entity
3. Add another entity

##### **Related issues**

None that I could locate

##### **Suggest a Fix**

There is a superflous comma in the generated entity.module.ts.  It should be removed.
```
{
        path: 'something',
        loadChildren: () => import('./something/something.module').then(m => m.JhipsterSomethingModule),
      },
      , <------ This one
      {
        path: 'something-else',
        loadChildren: () => import('./something-else/something-else.module').then(m => m.JhipsterSomethingElseModule),
      },
```

##### **JHipster Version(s)**

6.9.0

##### **JHipster configuration**

<!--
To provide all information we need, you should run `jhipster info` in the project root folder, and
copy/paste the result here.
The `.yo-rc.json` file generated in the root folder is mandatory for bug reports. This will help us to replicate the scenario.
You should remove any sensitive information like the rememberMe key or the jwtSecretKey key.
---
If you have a JDL please wrap it in below structure
  <details>
  <summary>JDL definitions</summary>
  <pre>
     JDL content here
  </pre>
  </details>
-->

```
INFO! Using JHipster version installed locally in current project's node_modules
INFO! Executing jhipster:info
INFO! Options: from-cli: true
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
jhipster@0.0.1-SNAPSHOT /home/schindlerj/src/temp/jhipster/upstream/angular-multi-entity
└── generator-jhipster@6.9.0 

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "6.9.0",
    "applicationType": "monolith",
    "baseName": "jhipster",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "embeddableLaunchScript": false,
    "useSass": true,
    "clientPackageManager": "npm",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "creationTimestamp": 1589918996227,
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en"],
    "blueprints": [],
    "lastLiquibaseTimestamp": 1589919632000
  }
}

</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Something {
  name String
}
entity SomethingElse {
  num Integer
}
relationship ManyToOne {
  SomethingElse{something(name)} to Something
}


</pre>
</details>


##### **Environment and Tools**

openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment 18.9 (build 11.0.7+10)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.7+10, mixed mode)

git version 2.25.4

node: v12.6.0

npm: 6.14.5

yeoman: 3.1.1

yarn: 1.22.4

Docker version 19.03.8, build afacb8b7f0

docker-compose version 1.25.4, build unknown

identical .jhipster/Something.json
identical .jhipster/SomethingElse.json
INFO! Congratulations, JHipster execution is complete!

```
##### **Browsers and Operating System**

Fedora Linux 31

-   [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/client/needle-api/needle-client-angular.js
+++ b/generators/client/needle-api/needle-client-angular.js
@@ -194,17 +194,18 @@ module.exports = class extends needleClientBase {
         this.addIcon(iconName);
     }
 
-    _addRoute(route, modulePath, moduleName, needleName, filePath, prefix = ',') {
+    _addRoute(route, modulePath, moduleName, needleName, filePath) {
         const isRouteAlreadyAdded = jhipsterUtils.checkStringInFile(filePath, `path: '${route}'`, this.generator);
         if (isRouteAlreadyAdded) {
             return;
         }
         const errorMessage = `${chalk.yellow('Route ') + route + chalk.yellow(` not added to ${filePath}.\n`)}`;
-        const routingEntry = this.generator.stripMargin(`${prefix}
-            |      {
+        const routingEntry = this.generator.stripMargin(
+            `{
             |        path: '${route}',
             |        loadChildren: () => import('${modulePath}').then(m => m.${moduleName})
-            |      }`);
+            |      },`
+        );
         const rewriteFileModel = this.generateFileModel(filePath, needleName, routingEntry);
         this.addBlockContentToFile(rewriteFileModel, errorMessage);
     }
@@ -220,15 +221,12 @@ module.exports = class extends needleClientBase {
 
             if (!isSpecificEntityAlreadyGenerated) {
                 const appName = this.generator.getAngularXAppName();
-                const isAnyEntityAlreadyGenerated = jhipsterUtils.checkStringInFile(entityModulePath, 'loadChildren', this.generator);
-                const prefix = isAnyEntityAlreadyGenerated ? ',' : '';
-
                 const modulePath = `./${entityFolderName}/${entityFileName}.module`;
                 const moduleName = microServiceName
                     ? `${this.generator.upperFirstCamelCase(microServiceName)}${entityAngularName}Module`
                     : `${appName}${entityAngularName}Module`;
 
-                this._addRoute(entityUrl, modulePath, moduleName, 'jhipster-needle-add-entity-route', entityModulePath, prefix);
+                this._addRoute(entityUrl, modulePath, moduleName, 'jhipster-needle-add-entity-route', entityModulePath);
             }
         } catch (e) {
             this.generator.debug('Error:', e);

--- a/generators/client/templates/angular/src/main/webapp/app/admin/admin-routing.module.ts.ejs
+++ b/generators/client/templates/angular/src/main/webapp/app/admin/admin-routing.module.ts.ejs
@@ -71,7 +71,7 @@ import { RouterModule } from '@angular/router';
         {
           path: 'metrics',
           loadChildren: () => import('./metrics/metrics.module').then(m => m.MetricsModule)
-        }
+        },
         /* jhipster-needle-add-admin-route - JHipster will add admin routes here */
       ]
     )

--- a/test/needle-api/needle-client-angular.spec.js
+++ b/test/needle-api/needle-client-angular.spec.js
@@ -178,9 +178,9 @@ describe('needle API Angular: JHipster client generator with blueprint', () => {
         assert.fileContent(
             `${CLIENT_MAIN_SRC_DIR}app/entities/entity.module.ts`,
             '      {\n' +
-                "        path: 'entityUrl',\n" +
-                "        loadChildren: () => import('./entityFolderName/entityFileName.module').then(m => m.MicroServiceNameentityNameModule)\n" +
-                '      }'
+            "        path: 'entityUrl',\n" +
+            "        loadChildren: () => import('./entityFolderName/entityFileName.module').then(m => m.MicroServiceNameentityNameModule)\n" +
+            '      }'
         );
     });
 
@@ -195,11 +195,11 @@ describe('needle API Angular: JHipster client generator with blueprint', () => {
     it('admin module contains the routing added by needle api', () => {
         assert.fileContent(
             `${CLIENT_MAIN_SRC_DIR}app/admin/admin-routing.module.ts`,
-            ',\n' +
-                '      {\n' +
-                "        path: 'entity-audit',\n" +
-                "        loadChildren: () => import('./entity-audit/entity-audit.module').then(m => m.EntityAuditModule)\n" +
-                '      }'
+            '        },\n' +
+            '        {\n' +
+            "        path: 'entity-audit',\n" +
+            "        loadChildren: () => import('./entity-audit/entity-audit.module').then(m => m.EntityAuditModule)\n" +
+            '      },'
         );
     });
 

--- a/test/needle-api/needle-client-angular.spec.js
+++ b/test/needle-api/needle-client-angular.spec.js
@@ -178,9 +178,9 @@ describe('needle API Angular: JHipster client generator with blueprint', () => {
         assert.fileContent(
             `${CLIENT_MAIN_SRC_DIR}app/entities/entity.module.ts`,
             '      {\n' +
-            "        path: 'entityUrl',\n" +
-            "        loadChildren: () => import('./entityFolderName/entityFileName.module').then(m => m.MicroServiceNameentityNameModule)\n" +
-            '      }'
+                "        path: 'entityUrl',\n" +
+                "        loadChildren: () => import('./entityFolderName/entityFileName.module').then(m => m.MicroServiceNameentityNameModule)\n" +
+                '      }'
         );
     });
 
@@ -196,10 +196,10 @@ describe('needle API Angular: JHipster client generator with blueprint', () => {
         assert.fileContent(
             `${CLIENT_MAIN_SRC_DIR}app/admin/admin-routing.module.ts`,
             '        },\n' +
-            '        {\n' +
-            "        path: 'entity-audit',\n" +
-            "        loadChildren: () => import('./entity-audit/entity-audit.module').then(m => m.EntityAuditModule)\n" +
-            '      },'
+                '        {\n' +
+                "        path: 'entity-audit',\n" +
+                "        loadChildren: () => import('./entity-audit/entity-audit.module').then(m => m.EntityAuditModule)\n" +
+                '      },'
         );
     });
 

